src/o365/yammer/commands/message/message-remove.ts

```diff
@@ -0,0 +1,94 @@
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import YammerCommand from "../../../base/YammerCommand";
+import request from '../../../../request';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: number;
+}
+
+class YammerMessageRemoveCommand extends YammerCommand {
+  public get name(): string {
+    return `${commands.YAMMER_MESSAGE_REMOVE}`;
+  }
+
+  public get description(): string {
+    return 'Removes a Yammer message';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const requestOptions: any = {
+      url: `${this.resource}/v1/messages/${args.options.id}.json`,
+      headers: {
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json;odata=nometadata'
+      },
+      json: true
+    };
+
+    request
+      .delete(requestOptions)
+      .then((res: any): void => {
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+        
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '--id <id>',
+        description: 'The id of the Yammer message'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.id) {
+        return 'Required id value is missing';
+      }
+      if (typeof args.options.id !== 'number') {
+        return `${args.options.id} is not a number`;
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  Remarks:
+  
+    ${chalk.yellow('Attention:')} In order to use this command, you need to grant the Azure AD
+    application used by the Office 365 CLI the permission to the Yammer API.
+    To do this, execute the ${chalk.blue('consent --service yammer')} command.
+
+    To remove a message, you must either (1) have posted the message yourself (2) be an administrator of the group the message was posted to or (3) be an admin of the network the message is in.
```

Comment:
I would suggest (1), (2), (3) to be new lines. It is sometimes hard to read the long sentence lines on small cmd windows/prompts opened.